### PR TITLE
refactor(router_core): move hardcoded value to constant for max invitations (fix #5938)

### LINKED ISSUE #5938: [REFACTOR] Move Hardcoded Value to Constant for Max Invitations
### Feature Description

Here:
https://github.com/juspay/hyperswitch/blob/b2364414edab6d44e276f2133bc988610693f155/crates/router/src/core/user.rs#L501

In the current implementation, the value 10 is hardcoded when checking the number of invite requests in the code. To improve maintainability and avoid magic numbers, this value should be moved to a constant in crates/router/src/consts/user.rs.

### Possible Implementation

- Create a new constant in crates/router/src/consts/user.rs to represent the maximum number of allowed invite requests.
- Refactor the code to replace the hardcoded value 10 with this new constant.

### Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### Are you willing to submit a PR?

No, but I'm happy to collaborate on a PR with someone else

### PR DESCRIPTION
**Pull request by ramkumargs**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/router/src/core/user.rs [hunk 1]
```diff
@@ -498,7 +498,7 @@ pub async fn invite_multiple_user(
     req_state: ReqState,
     auth_id: Option<String>,
 ) -> UserResponse<Vec<InviteMultipleUserResponse>> {
-    if requests.len() > 10 {
+    if requests.len() > consts::user::MAX_INVITE_REQUESTS {
         return Err(report!(UserErrors::MaxInvitationsError))
             .attach_printable("Number of invite requests must not exceed 10");
```
Comment (ThisIsMani) [inline crates/router/src/core/user.rs:503]: Change this `attach_printable` as well.

Try to log the number of requests as well in this.

**Review by ThisIsMani — CHANGES_REQUESTED** [review]

File: crates/router/src/core/user.rs (hunk 1 above)
Comment (ramkumargs) [reply crates/router/src/core/user.rs:503]: Changes are done and committed the code.

**Comment by ramkumargs** [comment]
@ThisIsMani, Changes are done and committed the code.

**Review by ThisIsMani — APPROVED** [review]

**Comment by ThisIsMani** [comment]
@ramkumargs, were you able to test this?

**Review by apoorvdixit88 — APPROVED** [review]

**Comment by ThisIsMani** [comment]
> @ramkumargs, were you able to test this?
Let me know if you need instructions on this.

**Comment by ramkumargs** [comment]
> > @ramkumargs, were you able to test this?
> 
> Let me know if you need instructions on this.

@ThisIsMani please give me the steps for testing.

**Comment by ramkumargs** [comment]
> Hey @ramkumargs, sorry for the late reply.
> 
> Steps for testing:
> 
> 1. Create a user
>    ```
>    curl --location 'http://localhost:8080/user/signup' \
>    --header 'Content-Type: application/json' \
>    --header 'api-key: test_admin' \
>    --data '{
>        "email": "email",
>        "password": "password"
>    }'
>    ```
> 2. The above API will give a token in the response.
>    ```
>    curl --location 'http://localhost:8080/user/2fa/terminate?skip_two_factor_auth=true' \
>    --header 'Authorization: Bearer {token from signup}'
>    ```
> 3. The above API will give one more token. With that you can hit invite API.
>    ```
>    curl --location 'http://localhost:8080/user/user/invite_multiple' \
>    --header 'Content-Type: application/json' \
>    --header 'Authorization: Bearer {token from 2FA API}' \
>    --data-raw '[
>        {
>            "email": "email",
>            "name": "name",
>            "role_id": "merchant_view_only"
>        },
>        {
>            "email": "email2",
>            "name": "name2",
>            "role_id": "merchant_view_only"
>        }
>    ]'
>    ```
>    
>    
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    
>    * If the number of elements in the requests are more than what you configured, it should throw error.
> 
> Let me know if you face any issues. If you were able to check, please attach a screenshot in the description.

@ThisIsMani , I will update you after testing.

**Comment by ramkumargs** [comment]
> Hey @ramkumargs, sorry for the late reply.
> 
> Steps for testing:
> 
> 1. Create a user
>    ```
>    curl --location 'http://localhost:8080/user/signup' \
>    --header 'Content-Type: application/json' \
>    --header 'api-key: test_admin' \
>    --data '{
>        "email": "email",
>        "password": "password"
>    }'
>    ```
> 2. The above API will give a token in the response.
>    ```
>    curl --location 'http://localhost:8080/user/2fa/terminate?skip_two_factor_auth=true' \
>    --header 'Authorization: Bearer {token from signup}'
>    ```
> 3. The above API will give one more token. With that you can hit invite API.
>    ```
>    curl --location 'http://localhost:8080/user/user/invite_multiple' \
>    --header 'Content-Type: application/json' \
>    --header 'Authorization: Bearer {token from 2FA API}' \
>    --data-raw '[
>        {
>            "email": "email",
>            "name": "name",
>            "role_id": "merchant_view_only"
>        },
>        {
>            "email": "email2",
>            "name": "name2",
>            "role_id": "merchant_view_only"
>        }
>    ]'
>    ```
>    
>    
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    
>    * If the number of elements in the requests are more than what you configured, it should throw error.
> 
> Let me know if you face any issues. If you were able to check, please attach a screenshot in the description.

@ThisIsMani , I will test and update you today

**Comment by gorakhnathy7** [comment]
Hey @ramkumargs Any update here?

**Comment by ramkumargs** [comment]
Hi @gorakhnathy7 sorry for late reply. I was struck on my some personal issue.
I have started the testing now.

**Comment by ramkumargs** [comment]
Hi @gorakhnathy7 sorry for late reply. I was struck on my some personal issue.
I have started the testing now.

**Comment by ramkumargs** [comment]
Hi,
When running below command, hanging at => **[hyperswitch-web 5/5] RUN npm install**
docker compose up -d (Expecting to run this command before executing curl commands)
image
Am I proceeding in the right way for testing?

**Comment by ramkumargs** [comment]
> curl --location 'http://localhost:8080/user/signup' \
> --header 'Content-Type: application/json' \
> --header 'api-key: test_admin' \
> --data '{
>     "email": "email",
>     "password": "password"
> }'

Not started. Failing at hyperswitch-hyperswitch-server-1, after long build time
image

**Comment by ThisIsMani** [comment]
Can you use `cargo run` instead.

**Pull request closed without merging**